plugins/network-elements/nuage-vsp/src/com/cloud/network/element/NuageVspElement.java

```diff
@@ -196,6 +227,20 @@ public boolean implement(Network network, NetworkOffering offering, DeployDestin
             ResourceUnavailableException, InsufficientCapacityException {
         s_logger.debug("Entering NuageElement implement function for network " + network.getDisplayText() + " (state " + network.getState() + ")");
```

Comment:
Wrap in a ``if (s_logger.isDebugEnabled()) { ... }`` block to avoid unnecessary pressure on the string pool when debug logging is not enabled.